lib/rules/no-invalid-html-attribute.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * @fileoverview Check if tag attributes to have non-valid value
+ * @author Sebastian Malton
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+/**
+ * Map between attributes and a mapping between valid values and a set of tags they are valid on
+ * @type {Map<string, Map<string, Set<string>>>}
+ */
+const VALID_VALUES = new Map();
+VALID_VALUES.set('rel', new Map())
+  .get('rel')
+  .set('alternate', new Set(['link', 'area', 'a']))
+  .set('author', new Set(['link', 'area', 'a']))
+  .set('bookmark', new Set(['area', 'a']))
+  .set('canonical', new Set(['link']))
+  .set('dns-prefetch', new Set(['link']))
+  .set('external', new Set(['area', 'a', 'form']))
+  .set('help', new Set(['link', 'area', 'a', 'form']))
+  .set('icon', new Set(['link']))
+  .set('license', new Set(['link', 'area', 'a', 'form']))
+  .set('manifest', new Set(['link']))
+  .set('modulepreload', new Set(['link']))
+  .set('next', new Set(['link', 'area', 'a', 'form']))
+  .set('nofollow', new Set(['area', 'a', 'form']))
+  .set('noopener', new Set(['area', 'a', 'form']))
+  .set('noreferrer', new Set(['area', 'a', 'form']))
+  .set('opener', new Set(['area', 'a', 'form']))
+  .set('pingback', new Set(['link']))
+  .set('preconnect', new Set(['link']))
+  .set('prefetch', new Set(['link']))
+  .set('preload', new Set(['link']))
+  .set('prerender', new Set(['link']))
+  .set('prev', new Set(['link', 'area', 'a', 'form']))
+  .set('search', new Set(['link', 'area', 'a', 'form']))
+  .set('stylesheet', new Set(['link']))
+  .set('tag', new Set(['area', 'a']));
+
+/**
+* Map between attributes and set of tags that the attribute is valid on
+* @type {Map<string, Set<string>>}
+*/
+const COMPONENT_ATTRIBUTE_MAP = new Map();
+COMPONENT_ATTRIBUTE_MAP.set('rel', new Set(['link', 'a', 'area', 'form']));
+
+function splitIntoRangedParts(node, regex) {
+  const res = [];
+  const valueRangeStart = node.range[0] + 1; // the plus one is for the initial quote
+  let match;
+
+  // eslint-disable-next-line no-cond-assign
+  while ((match = regex.exec(node.value)) !== null) {
```

Comment:
please use https://npmjs.com/string.prototype.matchall and `Array.from` for this rather than a while loop.